code/modules/bitrunning/objects/quantum_server.dm

```diff
@@ -0,0 +1,739 @@
+	var/servo_bonus = 0
+	/// The turfs we can place a hololadder on.
+	var/turf/exit_turfs = list()
+	/// The turfs on station where we generate loot.
+	var/turf/receive_turfs = list()
+
+/obj/machinery/quantum_server/Initialize(mapload)
+	. = ..()
+
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/quantum_server/LateInitialize()
+	. = ..()
+
+	if(isnull(console_ref))
+		find_console()
+
+	RegisterSignals(src, list(COMSIG_MACHINERY_BROKEN, COMSIG_MACHINERY_POWER_LOST), PROC_REF(on_broken))
+	RegisterSignal(src, COMSIG_QDELETING, PROC_REF(on_delete))
+	RegisterSignal(src, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
+	RegisterSignal(src, COMSIG_BITRUNNER_CLIENT_CONNECTED, PROC_REF(on_client_connected))
+	RegisterSignal(src, COMSIG_BITRUNNER_CLIENT_DISCONNECTED, PROC_REF(on_client_disconnected))
+	RegisterSignal(src, COMSIG_BITRUNNER_COP_SPAWNED, PROC_REF(on_threat_created))
+
+	// This further gets sorted in the client by cost so it's random and grouped
+	available_domains = shuffle(subtypesof(/datum/lazy_template/virtual_domain))
+
+/obj/machinery/quantum_server/Destroy(force)
+	. = ..()
+
+	available_domains.Cut()
+	QDEL_LIST(mutation_candidates)
+	QDEL_LIST(occupant_mind_refs)
+	QDEL_LIST(spawned_threats)
+	QDEL_NULL(exit_turfs)
+	QDEL_NULL(receive_turfs)
+	QDEL_NULL(generated_domain)
+	QDEL_NULL(generated_safehouse)
+
+/obj/machinery/quantum_server/update_appearance(updates)
+	if(isnull(generated_domain) || !is_operational)
+		set_light(0)
+		return ..()
+
+	set_light_color(cooling_off ? LIGHT_COLOR_FIRE : LIGHT_COLOR_BABY_BLUE)
+	set_light(2, 1.5)
+
+	return ..()
+
+/obj/machinery/quantum_server/update_icon_state()
+	if(isnull(generated_domain) || !is_operational)
+		icon_state = base_icon_state
+		return ..()
+
+	icon_state = "[base_icon_state]_[cooling_off ? "off" : "on"]"
+	return ..()
+
+/obj/machinery/quantum_server/crowbar_act(mob/living/user, obj/item/crowbar)
+	. = ..()
+
+	if(!get_is_ready())
+		balloon_alert(user, "it's scalding hot!")
+		return TRUE
+	if(length(occupant_mind_refs))
+		balloon_alert(user, "all clients must disconnect!")
+		return TRUE
+	if(default_deconstruction_crowbar(crowbar))
+		return TRUE
+	return FALSE
+
+/obj/machinery/quantum_server/screwdriver_act(mob/living/user, obj/item/screwdriver)
+	. = ..()
+
+	if(!get_is_ready())
+		balloon_alert(user, "it's scalding hot!")
+		return TRUE
+	if(default_deconstruction_screwdriver(user, "[base_icon_state]_panel", icon_state, screwdriver))
+		return TRUE
+	return FALSE
+
+/obj/machinery/quantum_server/RefreshParts()
+	. = ..()
+
+	var/capacitor_rating = 1.15
+	var/datum/stock_part/capacitor/cap = locate() in component_parts
+	capacitor_rating -= cap.tier * 0.15
+
+	capacitor_coefficient = capacitor_rating
+
+	var/datum/stock_part/scanning_module/scanner = locate(/datum/stock_part/scanning_module) in component_parts
+	if(scanner)
+		scanner_tier = scanner.tier
+
+	var/servo_rating = 0
+	for(var/datum/stock_part/servo/servo in component_parts)
```

Comment:
```suggestion
	for(var/datum/stock_part/servo/servo as anything in component_parts)
```
This is an optimization to prevent typechecking.

Reply:
It's not only servos in component parts tho 🤔